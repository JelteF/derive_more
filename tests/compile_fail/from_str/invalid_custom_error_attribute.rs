#[derive(derive_more::FromStr)]
#[from_str(error(CustomError, CustomError::new, CustomError::from))]
pub enum Foo {
    Bar,
}

#[derive(derive_more::FromStr)]
#[from_str(error(CustomError, CustomError::from, CustomError::new))]
struct Baz;

#[derive(derive_more::FromStr)]
#[from_str(error())]
struct Empty;

fn main() {}
