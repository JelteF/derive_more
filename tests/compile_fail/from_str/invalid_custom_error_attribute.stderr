error: no more arguments allowed here
 --> tests/compile_fail/from_str/invalid_custom_error_attribute.rs:2:47
  |
2 | #[from_str(error(CustomError, CustomError::new, CustomError::from))]
  |                                               ^

error: no more arguments allowed here
 --> tests/compile_fail/from_str/invalid_custom_error_attribute.rs:8:48
  |
8 | #[from_str(error(CustomError, CustomError::from, CustomError::new))]
  |                                                ^

error: unexpected end of input, expected identifier
  --> tests/compile_fail/from_str/invalid_custom_error_attribute.rs:12:18
   |
12 | #[from_str(error())]
   |                  ^
